solution/1800-1899/1891.Cutting Ribbons/Solution.rs

```diff
@@ -0,0 +1,20 @@
+fn maxLength(ribbons: Vec<i32>, k: i32) -> i32 {
+    let mut left = 0i32;
+    let mut right = *ribbons.iter().max().unwrap();
+    while left < right {
+        let mid = (left + right + 1) / 2;
+        let mut cnt = 0i32;
+        for &entry in ribbons.iter() {
+            cnt += entry / mid;
+            if cnt >= k {
+                break;
+            }
+        }
+        if cnt >= k {
+            left = mid;
+        } else {
+            right = mid - 1;
+        }
+    }
+    return left;
+}
```

Comment:
```suggestion
impl Solution {
    pub fn max_length(ribbons: Vec<i32>, k: i32) -> i32 {
        let mut left = 0i32;
        let mut right = *ribbons.iter().max().unwrap();
        while left < right {
            let mid = (left + right + 1) / 2;
            let mut cnt = 0i32;
            for &entry in ribbons.iter() {
                cnt += entry / mid;
                if cnt >= k {
                    break;
                }
            }
            if cnt >= k {
                left = mid;
            } else {
                right = mid - 1;
            }
        }
        return left;
    }
}
```

Reply:
@methanoya 

Hello, please implement based on the Rust code template from LeetCode, and add the complete code snippet to the files Solution.rs, README.md, and README_EN.md.

You can refer to this pull request: https://github.com/doocs/leetcode/pull/1802/files.

Thanks for your contribution :)